File: app/src/main/java/com/example/pam_app/repository/BucketRepository.java
package com.example.pam_app.repository;

import com.example.pam_app.model.Bucket;
import com.example.pam_app.model.BucketEntry;

import java.util.List;

import io.reactivex.Flowable;
import io.reactivex.Single;

public interface BucketRepository {
    Flowable<List<Bucket>> getList();

    void create(final Bucket bucket);

    void delete(final int id);

    Flowable<Bucket> get(final int id);

    Single<Bucket> get(final String title);

    Flowable<List<BucketEntry>> getEntryList();

    Flowable<List<Double>> getSpendingTotal();

    void addEntry(final BucketEntry entry, final int bucketId);

    void removeEntry(final BucketEntry entry, final int idBucket);

    Flowable<List<String>> getTitleListByType(final int type);
}

File: app/src/main/java/com/example/pam_app/repository/BucketRepository.kt
package com.example.pam_app.repository

import com.example.pam_app.model.Bucket
import com.example.pam_app.model.BucketEntry
import io.reactivex.Flowable
import io.reactivex.Single
import java.util.*
import kotlin.jvm.Synchronized

interface BucketRepository {
    val list: Flowable<List<Bucket>>
    fun getList(type: Boolean, date: Date?): Flowable<List<Bucket?>>
    fun update(bucket: Bucket?): Single<Int?>
    fun create(bucket: Bucket?): Single<Long?>
    fun delete(id: Int): Single<Int?>
    operator fun get(id: Int): Flowable<Bucket>
    operator fun get(title: String?): Single<Bucket>
    val entryList: Flowable<List<BucketEntry>>
    val spendingTotal: Flowable<List<Double?>?>?
    fun addEntry(entry: BucketEntry, bucketId: Int): Single<Long?>
    fun removeEntry(entry: BucketEntry, idBucket: Int)
    fun getTitleListByType(type: Int): Flowable<List<String?>?>?
}